language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -143,6 +145,18 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
             return CompletableFuture.supplyAsync(() -> res);
         }
 
+        // Compile the project with the experimental flag if the client has it enabled
+        if (capabilities.getExperimentalCapabilities().isExperimentalCompilationEnabled()) {
+            BuildOptions buildOptions = BuildOptions.builder()
+                    .setOffline(CommonUtil.COMPILE_OFFLINE)
+                    .setSticky(true)
```

Comment:
Can't we pass only the additional options externally, if not, in case a change is required to the options, changes have to be done in different places?